ride_share_noor.rb

```diff
@@ -0,0 +1,192 @@
+# rides (arrays of hashes)
+# each single ride has its own hash
+########################################################
+# Step 3: Make the data structure!
+# # Setup the entire data structure:
+# # based off of the notes you have above, create the
+# # and manually write in data presented in rides.csv
+# # You should be copying and pasting the literal data
+# # into this data structure, such as "DR0004"
+# # and "3rd Feb 2016" and "RD0022"
+
+trips_data = [{
+    driver_id: "DR0001",
+    rides: [
+        {
+            date: "3rd Feb 2016",
+            cost: 10,
+            rider_id: "RD0003",
+            rating: 3
+        },{
+            date: "3rd Feb 2016",
+            cost: 30,
+            rider_id: "RD0015",
+            rating: 4
+        },{
+            date: "5th Feb 2016",
+            cost: 45,
+            rider_id: "RD0003",
+            rating: 2
+        }]},{
+    driver_id: "DR0002",
+    rides: [
+        {
+            date: "3rd Feb 2016",
+            cost: 25,
+            rider_id: "RD0073",
+            rating: 5
+        },{
+            date: "4th Feb 2016",
+            cost: 15,
+            rider_id: "RD0013",
+            rating: 1
+        },{
+            date: "5th Feb 2016",
+            cost: 35,
+            rider_id: "RD0066",
+            rating: 3
+        }]},{
+    driver_id: "DR0003",
+    rides: [
+        {
+            date: "4th Feb 2016",
+            cost: 5,
+            rider_id: "RD0066",
+            rating: 5
+        },{
+            date: "5th Feb 2016",
+            cost: 50,
+            rider_id: "RD0003",
+            rating: 2
+        }]},{
+    driver_id: "DR0004",
+    rides: [
+        {
+            date: "4th Feb 2016",
+            cost: 10,
+            rider_id: "RD0022",
+            rating: 4
+        },{
+            date: "5th Feb 2016",
+            cost: 20,
+            rider_id: "RD0073",
+            rating: 5
+        },{
+            date: "3rd Feb 2016",
+            cost: 5,
+            rider_id: "RD0022",
+            rating: 5
+        }]}]
+
+def line
+        puts "-" * 55
+end
+
+def line2
+        puts "-" * 104
+end
+########################################################
+# Step 4: Total Driver's Earnings and Number of Rides
+
+# Use an iteration blocks to print the following answers:
+# - the number of rides each driver has given
+
+line
+puts "Number of rides each driver has given"
+line
+trips_data.map { |driver_trips|
+  puts "Driver with ID of #{driver_trips[:driver_id]} made #{driver_trips[:rides].length} rides"}
+line
+
+# - the total amount of money each driver has made
+puts "Total amount of money each driver has made"
+line
+
+trips_data.map do |driver_trips|
+  sum = driver_trips[:rides].sum do |ride|
+    ride[:cost]
+  end
+  puts "Driver with ID of #{driver_trips[:driver_id]} earned $#{sum} "
+  end
+line
+
+# - the average rating for each driver
+puts "Average rating for each driver"
+line
+
+trips_data.map do |driver_trips|
```

Comment:
Similar issue to the above, when using `.map` you want to use the returned value and not do `puts`.  If you use `puts` just do `.each`.